backend/users/views.py

```diff
@@ -2,20 +2,26 @@
 from knox.views import LoginView as KnoxLoginView
 
 from rest_framework import viewsets, permissions, generics
-from rest_framework.permissions import AllowAny
+from rest_framework.permissions import AllowAny, IsAuthenticated
 from rest_framework.response import Response
 from rest_framework.authtoken.serializers import AuthTokenSerializer
 from django.contrib.auth import login
 
 from .mixins import StaffEditorPermissionMixin
 from .models import User
 from .serializers import UserSerializer, RegisterSerializer
+from users.serializers import UserSerializer
+
+from knox.auth import TokenAuthentication
+
 
 class UserViewSet(StaffEditorPermissionMixin, viewsets.ModelViewSet):
-    permission_class = [permissions.IsAuthenticated]
     queryset = User.objects.all()
     serializer_class = UserSerializer
 
+    def list(self, request, *args, **kwargs):
```

Comment:
what should be done in case we want the list API to be public (unauthenticated? )